streamz/tests/test_sources.py

```diff
@@ -0,0 +1,81 @@
+import pytest
+from streamz import Source
+from streamz.utils_test import wait_for, await_for, gen_test
+from tornado import gen
+import socket
+import time
+
+
+def test_tcp():
+    port = 9876
+    s = Source.from_tcp(port)
+    out = s.sink_to_list()
+    s.start()
+    time.sleep(0.02)
+
+    try:
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.connect(("localhost", port))
+        sock.send(b'data\n')
+        sock.close()
+
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.connect(("localhost", port))
+        sock.send(b'data\n')
+
+        sock2 = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock2.connect(("localhost", port))
+        sock2.send(b'data2\n')
+    finally:
+        s.stop()
+
+    wait_for(lambda: out == [b'data\n', b'data\n', b'data2\n'], 2, period=0.01)
+
+
+@gen_test(timeout=60)
+def test_tcp_async():
+    port = 9876
+    s = Source.from_tcp(port)
+    out = s.sink_to_list()
+    s.start()
+    yield gen.sleep(0.02)
+
+    try:
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.connect(("localhost", port))
+        sock.send(b'data\n')
+        sock.close()
+
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.connect(("localhost", port))
+        sock.send(b'data\n')
+
+        sock2 = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock2.connect(("localhost", port))
+        sock2.send(b'data2\n')
+        yield await_for(lambda: out == [b'data\n', b'data\n', b'data2\n'], 2,
+                        period=0.01)
+    finally:
+        s.stop()
+
+
+def test_http():
+    requests = pytest.importorskip('requests')
+    port = 9875
+    s = Source.from_http_server(port)
+    out = s.sink_to_list()
+    s.start()
+    time.sleep(0.02)  # allow loop to run
```

Comment:
Is this necessary?  If so, why this time?

In general, I'll probably challenge any sleep like this.  If it's necessary to make things run then it's likely too short in all cases (GC on travis can take arbitrarily long times) and so results in intermittent failures.  I spend a non-trivial time tracking these things down in the dask/distributed codebase, so I've grown fairly allergic to them.